lib/hexpm/repository/release.ex

```diff
@@ -92,13 +121,17 @@ defmodule Hexpm.Repository.Release do
   defp editable_error_message(:delete),
     do: "can only delete a release up to one hour after creation"
 
-  defp editable?(release) do
+  defp editable?(release, replace?) do
     is_nil(release.inserted_at) or
-      not release.package.repository.public or
+      private_repo_and_replaceable?(release, replace?) or
       within_seconds?(release.inserted_at, @one_hour) or
       within_seconds?(release.package.inserted_at, @one_day)
   end
 
+  defp private_repo_and_replaceable?(release, replace?) do
+    not release.package.repository.public and replace?
```

Comment:
oh, i was going thru the old convo about it and couldnt recall if we decided to do public packages as well. I think it is a good idea, but perhaps it is worth exiting with a specific message about needing the flag to avoid confusion.